sec_certs/sample/common_criteria.py

```diff
@@ -651,8 +705,129 @@ def compute_heuristics_cert_lab(self):
+        start_year = 1996
+        limit_year = datetime.now().year + 1
+        bsi_parts = cert_id.split("-")
+        cert_num = None
+        cert_version = None
+        cert_year = None
+
+        if len(bsi_parts) > 3:
+            cert_num = bsi_parts[3]
+        if len(bsi_parts) > 4:
+            if bsi_parts[4].startswith("V") or bsi_parts[4].startswith("v"):
+                cert_version = bsi_parts[4].upper()  # get version in uppercase
+            else:
+                cert_year = bsi_parts[4]
+        if len(bsi_parts) > 5:
+            cert_year = bsi_parts[5]
+
+        if cert_year is None:
+            for year in range(start_year, limit_year):
+                cert_id_possible = cert_id + "-" + str(year)
+
+                if cert_id_possible in all_cert_ids:
+                    # we found version with year
+                    cert_year = str(year)
+                    break
+
+        # reconstruct BSI number again
+        new_cert_id = "BSI-DSZ-CC"
+        if cert_num is not None:
+            new_cert_id += "-" + cert_num
+        if cert_version is not None:
+            new_cert_id += "-" + cert_version
+        if cert_year is not None:
+            new_cert_id += "-" + cert_year
+
+        return new_cert_id
+
+    @staticmethod
+    def _is_spain_cert_id(cert_id: str) -> bool:
+        return "-INF-" in cert_id
+
+    @staticmethod
+    def _fix_spain_cert_id(cert_id: str) -> str:
+        spain_parts = cert_id.split("-")
+        cert_year = spain_parts[0]
+        cert_batch = spain_parts[1]
+        cert_num = spain_parts[3]
+
+        if "v" in cert_num:
+            cert_num = cert_num[: cert_num.find("v")]
+        if "V" in cert_num:
+            cert_num = cert_num[: cert_num.find("V")]
+
+        new_cert_id = f"{cert_year}-{cert_batch}-INF-{cert_num}"  # drop version
+
+        return new_cert_id
+
+    @staticmethod
+    def _is_ocsi_cert_id(cert_id: str) -> bool:
+        return "OCSI/CERT" in cert_id
+
+    @staticmethod
+    def _fix_ocsi_cert_id(cert_id: str) -> str:
+        new_cert_id = cert_id
+        if not new_cert_id.endswith("/RC"):
+            new_cert_id = cert_id + "/RC"
+
+        return new_cert_id
+
+    def normalize_cert_id(self, all_cert_ids: Set[str]) -> None:
+        if self.heuristics.cert_id is None:
+            return None
+
+        cert_id = self.heuristics.cert_id.strip()
+        fixed_cert_id = cert_id
+
+        if CommonCriteriaCert._is_anssi_cert(cert_id):
+            fixed_cert_id = CommonCriteriaCert._fix_anssi_cert_id(cert_id)
+
+        if CommonCriteriaCert._is_bsi_cert(cert_id):
+            fixed_cert_id = CommonCriteriaCert._fix_bsi_cert_id(cert_id, all_cert_ids)
+
+        if CommonCriteriaCert._is_spain_cert_id(cert_id):
+            fixed_cert_id = CommonCriteriaCert._fix_spain_cert_id(cert_id)
+
+        if CommonCriteriaCert._is_ocsi_cert_id(cert_id):
+            fixed_cert_id = CommonCriteriaCert._fix_ocsi_cert_id(cert_id)
```

Comment:
Nitpicking: I would consider refactoring `_is_bsi_cert` and similar method to a single method: `get_cert_laboratory()` that would return `anssi`, `bsi`, `canada`, ... `unknown`. You could then define

```python
fix_method = {
    'anssi': CommonCriteriaCert._fix_anssi_cert_id
    'bsi': 'CommonCriteriaCert._fix_bsi_cert_id # <-- careful, you must actually use functools.partial here
    # and so on, you know what I mean
}
fix_method[self.get_cert_laboratory()](cert_id)
```

This is better expandable if we add something in the future, and it also has a single interface.